integration-tests/src/tests/client/sharding_upgrade.rs

```diff
@@ -293,9 +296,8 @@ impl TestShardUpgradeEnv {
 
         // Make sure that catchup is done before the end of each epoch, but when it is done is
         // by chance. This simulates when catchup takes a long time to be done
-        // Note: if the catchup happens only at the last block of an epoch then
-        // client will fail to produce the chunks in the first block of the next epoch.
-        let should_catchup = rng.gen_bool(P_CATCHUP) || next_height % self.epoch_length == 0;
+        let should_catchup =
```

Comment:
Ideally, instead of having a random num here, we should have independent tests that start the catchup on the first block and the middle block of an epoch.

Reply:
Yeah, sounds good, but let's do it separately. I'm a bit bored with those tests for now but if you feel like it I'll be happy to accept such a PR :)